Temporarily add fixes from the project that haven't been released yet:
https://github.com/mate-desktop/marco/commit/0f7a8c6d19874336040a11cf62a7ecb3c1dd9849
https://github.com/mate-desktop/marco/commit/81ca7129c159bdae1d9ae9a0cd4e196a8354e7ce
https://github.com/mate-desktop/marco/commit/dcd5d21e4c53f1670b33cec78c96bdd3da87037d

--- marco-1.26.1/src/core/display-private.h.orig	2022-11-12 17:47:39.000000000 +0000
+++ marco-1.26.1/src/core/display-private.h	2023-01-19 18:25:18.599029322 +0000
@@ -80,6 +80,7 @@
 struct _MetaDisplay {
 	char* name;
 	Display* xdisplay;
+    gboolean have_xres; /* Got XRES >= 1.2 */
 
 	Window leader_window;
 	Window timestamp_pinging_window;
--- marco-1.26.1/src/core/display.c.orig	2022-11-12 17:47:39.000000000 +0000
+++ marco-1.26.1/src/core/display.c	2023-01-19 18:29:28.327137994 +0000
@@ -50,6 +50,7 @@
 #include "compositor.h"
 #include <X11/Xatom.h>
 #include <X11/cursorfont.h>
+#include <X11/extensions/XRes.h>
 
 #ifdef HAVE_SOLARIS_XINERAMA
 	#include <X11/extensions/xinerama.h>
@@ -329,7 +330,8 @@
   Display *xdisplay;
   GSList *screens;
   GSList *tmp;
-  int i;
+  int i, event_base, error_base, major, minor;
+  gboolean have_xres = FALSE;
   guint32 timestamp;
 
   /* A list of all atom names, so that we can intern them in one go. */
@@ -343,6 +345,7 @@
   meta_verbose ("Opening display '%s'\n", XDisplayName (NULL));
 
   xdisplay = meta_ui_get_display ();
+  event_base = error_base = major = minor = 0;
 
   if (xdisplay == NULL)
     {
@@ -351,6 +354,14 @@
       return FALSE;
     }
 
+  /* Make sure to init Xres any extensions */
+  if (XResQueryExtension(xdisplay, &event_base, &error_base) &&
+      XResQueryVersion (xdisplay, &major, &minor) == 1)
+    {
+      if (major > 1 || (major == 1 && minor >= 2))
+          have_xres = TRUE;
+    }
+
   if (meta_is_syncing ())
     XSynchronize (xdisplay, True);
 
@@ -365,6 +376,7 @@
    */
   the_display->name = g_strdup (XDisplayName (NULL));
   the_display->xdisplay = xdisplay;
+  the_display->have_xres = have_xres;
   the_display->error_trap_synced_at_last_pop = TRUE;
   the_display->error_traps = 0;
   the_display->error_trap_handler = NULL;
--- marco-1.26.1/src/core/window-props.c.orig	2022-11-12 17:47:39.000000000 +0000
+++ marco-1.26.1/src/core/window-props.c	2023-01-19 18:32:29.342712724 +0000
@@ -306,14 +306,18 @@
 
   pid = -1;
 
+  if (window->display->have_xres == FALSE)
+    return pid;
+
   spec.client = window->xwindow;
   spec.mask = XRES_CLIENT_ID_PID_MASK;
 
-  XResQueryClientIds (window->display->xdisplay,
+  if (!XResQueryClientIds (window->display->xdisplay,
                       1,
                       &spec,
                       &num_ids,
-                      &client_ids);
+                      &client_ids))
+      return pid;
 
   for (i = 0; i < num_ids; i++)
     {
